test/functional/mempool_wallet_transactions.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
```

Comment:
There's some duplicated code between this and mempool_rebroadcast.py. Consider merging them.

Note that you're exercising this behaviour using the `sendrawtransaction` RPC so you don't even need to use a wallet here.